src/LagrangianParticleTracking/calculate_particle_tendency_contributions.jl

```diff
@@ -0,0 +1,84 @@
+function calculate_particle_tendency_contributions!(particles, model, active_properties)
+    arch = model.architecture
+
+    workgroup = min(length(particles), 256)
+    worksize = length(particles)
+    
+    Gp_kernel! = calculate_particle_tendency_kernel!(device(arch), workgroup, worksize)
+
+    events = []
+
+    for field in active_properties
+        property = getproperty(particles.properties, field.property)
+        tendency = model.timestepper.Gⁿ[field.tracer]
+
+        Gp_event = Gp_kernel!(property, tendency, particles, model.grid, dependencies = Event(device(arch)))
+
+        push!(events, Gp_event)
+    end
+
+    wait(device(arch), MultiEvent(Tuple(events)))
+
+    return nothing
+end
+
+calculate_particle_tendency_contributions!(model) = calculate_particle_tendency_contributions!(model.particles, model)
+calculate_particle_tendency_contributions!(particles::Nothing, model) = nothing
+calculate_particle_tendency_contributions!(particles, model) = !isnothing(particles.parameters) ? (:active_properties in keys(particles.parameters) ? calculate_particle_tendency_contributions!(particles, model, particles.parameters.active_properties) : nothing) : nothing
+calculate_particle_tendency_contributions!(particles, model, active_fields::Nothing) = nothing
+
+@inline function get_mask(x, y, z, grid, loc)
```

Comment:
What's the purpose of this function? Maybe add a docstring?

Reply:
It works out which grid points are the nearest to the particle (since particles can be at any arbitrary position rather than on grid points), and returns a tuple of tuples of (i, j, k, d) where d is the distance from the particle to point ijk, along with a normalization factor which is 1/sum(1/d), so the contribution you add to ijk is tendency*d_ijk*normalization factor.

I'll add a docstring

Reply:
how about calling it `nearest_cell_center` ?